tike/ptycho.py

```diff
@@ -178,28 +176,31 @@ def fast_pad(unpadded_grid, npadv, npadh):
     is faster because of the way that np.pad uses concatenate.
 
     """
+    assert npadv >= 0 and npadh >= 0, "Pads must be non-negative."
```

Comment:
If pad is zero, then you can just return unpadded_grid.

Reply:
I think this depends on the size of the probe support. We may want to pad for simulations, because we use a small circular compact probe compared to the detector. But for real data, the probe can be larger like a sink shaped function with main intensity at the center with decaying tails. Therefore, It may make sense to set the probe shape  the same as the detector shape.